[py-openslide-python] added verion 1.1.2

qwertos opened the pull request.

qwertos committed:
[py-openslide-python] added verion 1.1.2 and set max py-setuptools version for 1.1.1

tldahlgren reviewed (dismissed):
Confirmed sha256.  Leaving merge decision to @adamjstewart

adamjstewart commented on var/spack/repos/builtin/packages/py-openslide-python/package.py:
```diff
@@ -12,9 +12,12 @@ class PyOpenslidePython(PythonPackage):
     version('1.1.1', sha256='33c390fe43e3d7d443fafdd66969392d3e9efd2ecd5d4af73c3dbac374485ed5')
 
     depends_on('openslide@3.4.0:')
     depends_on('python@2.6:2.8,3.3:')
```
```suggestion
    depends_on('python@2.6:2.8,3.3:', type=('build', 'run'))
```
Presumably Python is also needed at run-time. May also be needed at link-time, you'll have to check

adamjstewart commented on var/spack/repos/builtin/packages/py-openslide-python/package.py:
```diff
@@ -12,9 +12,12 @@ class PyOpenslidePython(PythonPackage):
-    depends_on('py-setuptools', type='build')
+    # https://github.com/openslide/openslide-python/pull/76
+    depends_on('py-setuptools@:45', type='build', when="@1.1.1")
+    depends_on('py-setuptools',     type='build', when="@1.1.2")
```
```suggestion
    depends_on('py-setuptools',     type='build', when="@1.1.2:")
```

adamjstewart commented on var/spack/repos/builtin/packages/py-openslide-python/package.py:
```diff
@@ -12,9 +12,12 @@ class PyOpenslidePython(PythonPackage):
+    # https://github.com/openslide/openslide-python/pull/76
+    depends_on('py-setuptools@:45', type='build', when="@1.1.1")
+    depends_on('py-setuptools',     type='build', when="@1.1.2")
     depends_on('py-pillow-simd+jpeg+jpeg2000+tiff', type=('build', 'run'))
```
```suggestion
    depends_on('pil', type=('build', 'run'))
```
We have a virtual package for PIL that lets you choose between `py-pillow` and `py-pillow-simd`. The latter only builds on systems with SIMD, it doesn't work on Apple M1 for example. Unfortunately, virtual packages don't support variants yet.

qwertos committed:
[py-openslide-python]

- setuptools required for all possible newer versions
- python is type build run

qwertos committed:
[py-openslide-python] use pil provider
